crates/tarball/src/lib.rs

```diff
@@ -40,24 +40,21 @@ pub async fn download_and_extract(
     // For now: .pacquet/fast-querystring/1.0.0
     let extract_location = cache_directory.join(name).join(version);
 
-    let mut stream =
-        reqwest::get(url).await.map_err(|e| TarballError::Network(e))?.bytes_stream();
+    let mut stream = reqwest::get(url).await.map_err(TarballError::Network)?.bytes_stream();
 
-    let mut file = File::create(&tarball_location)
-        .map_err(|e| TarballError::Io(e))?;
+    let mut file = File::create(&tarball_location).map_err(TarballError::Io)?;
 
     while let Some(item) = stream.next().await {
-        let chunk = item.map_err(|e| TarballError::Network(e))?;
-        file.write_all(&chunk).map_err(|e| TarballError::Io(e))?;
+        let chunk = item.map_err(TarballError::Network)?;
+        file.write_all(&chunk).map_err(TarballError::Io)?;
```

Comment:
It's good that, I didn't ask this and patiently waited! Good to know. Similar concept exist in Javascript, Swift and even C++.